setupShot.py

```diff
@@ -0,0 +1,572 @@
+
+        # App definition for Shotgun requests
+        self.app = sgtk.platform.current_bundle()
+
+        self.entityContext = self.app.context.entity
+
+        # These colors come from the bdColors.nk file in /resources
+        self.backDropColors = [1833515007, 1433215999, 943285503, 1767926015, 1683365887,
+                               711221759, 1281640447, 1680491263, 656877567, 1679696383]
+
+        # Store a list of all SG statuses including ShortCode, Name & BGColor (Hex String)
+        self.statusList = self.app.shotgun.find('Status', [['code', 'is_not', '']], ['name', 'bg_color', 'code'])
+
+        # Populate the listbox on the UI with all output types found for the context entity
+        self._populateListWidget()
+
+        self.ui.toggleSelectBox.stateChanged.connect(lambda: self._toggleAllItems())
+
+        self.ui.buttonBox.accepted.connect(lambda: self._doImport())
+        self.ui.buttonBox.rejected.connect(lambda: self.close())
+
+    def _toggleAllItems(self):
+        state = self.ui.toggleSelectBox.checkState()
+
+        for i in range(0, self.ui.listWidget.count()):
+            self.ui.listWidget.item(i).setCheckState(state)
+
+    def _populateListWidget(self):
+        if self.entityContext is None:
+            QtGui.QMessageBox.information(
+                self,
+                "Error!",
+                "Failed to find context for this app to run\n\nTry opening a scene through SG",
+            )
+            return
+
+        contextID = self.entityContext['id']
+        entityType = self.entityContext['type']
+
+        self.theEntity = self.app.shotgun.find_one(entityType, [['id', 'is', contextID]], ['code', 'id'])
+        # self.theEntity = self.app.shotgun.find_one(entityType, [['id', 'is', 18144]], ['code', 'id'])
+
+        # Find all published files associated with the Entity the user is in
+        # We are only interested in PublishedFiles with a blank element field
+        # This filters out all thumbnails and editorial movies
+        filters = [
+            ['entity', 'is', self.theEntity],
+            ['sg_publishing_status', 'is', 'cmpt'],
+            ['sg_step_output', 'type_is_not', '']
+        ]
+        # Store these for later use, no reason to query SG twice
+        publishes = self.app.shotgun.find('PublishedFile', filters, ['code', 'sg_step_output', 'sg_publish_path', 'sg_multi_publish', 'sg_element',
+                                                                     'version.Version.sg_status_list',
+                                                                     'version.Version.sg_is_hero',
+                                                                     'version.Version.sg_primary_published_file.PublishedFile.id',
+                                                                     'task.Task.step.Step.code'])
+
+        idsToRemove = []  # List of publishes that are to be removed from self.publishes as we've found 'working' replacements

```

Comment:
there is no property self.publishes (maybe there was once). Make the comments match the actual code, otherwise they do more harm than good